File: project/FridgeBackend/Data/DTO/InventoryEvent.cs
using FridgeBackend.Data.Enums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FridgeBackend.Data.DTO;

public class InventoryEvent
{
    [Key] public Guid Id { get; set; } = Guid.NewGuid();
    [Required] public Guid ItemId { get; set; }
    [ForeignKey(nameof(ItemId))] public Item? Item { get; set; }
    public double DeltaQuantity { get; set; }
    public decimal? UnitPriceAtEvent { get; set; }
    public EventType Type { get; set; } = EventType.Adjust;
    public DateTime OccurredAt { get; set; } = DateTime.UtcNow;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: project/flutter_fridge_app/lib/models/inventory_event.dart
import 'package:flutter_fridge_app/models/Enums/inventory_event_type.dart';

class InventoryEvent {
  final String id;
  final String itemId;
  final double deltaQuantity;
  final double? unitPriceAtEvent;
  final InventoryEventType type;
  final DateTime occurredAt;
  final DateTime createdAt;

  const InventoryEvent({
    required this.id,
    required this.itemId,
    required this.deltaQuantity,
    this.unitPriceAtEvent,
    required this.type,
    required this.occurredAt,
    required this.createdAt,
  });
}
